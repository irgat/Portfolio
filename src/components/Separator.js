import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    label: PropTypes.string,
};

const defaultProps = {
    label: null,
};

const Separator = (props) => {
    return <h3>{props.label && props.label}</h3>;
};

Separator.propTypes = propTypes;
Separator.defaultProps = defaultProps;

export default Separator;