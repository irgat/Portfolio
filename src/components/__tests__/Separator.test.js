import React from 'react';
import Enzyme from 'enzyme';
import Separator from '../Separator';

const validData = [
    { label: 'copy' },
];

const mixedData = [
    {},
    { key: 'value' },
].concat(validData);

const separator = (props) => {
    return Enzyme.shallow(<Separator {...props} />);
};

describe('<Separator />', () => {
    it.each(mixedData)('renders a <h3>', (props) => {
        expect(separator(props).find('h3').getElements()).toMatchSnapshot();
    });

    it.each(validData)('renders label', (props) => {
        expect(separator(props).find('h3').text()).toMatch(props.label);
    });
});
