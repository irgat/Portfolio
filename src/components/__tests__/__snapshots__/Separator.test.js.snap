// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> renders a <h3> 1`] = `
Array [
  <h3 />,
]
`;

exports[`<Separator /> renders a <h3> 2`] = `
Array [
  <h3 />,
]
`;

exports[`<Separator /> renders a <h3> 3`] = `
Array [
  <h3>
    copy
  </h3>,
]
`;
